### ISSUE TITLE
Determine if device is compatible with DeviceOrientationControls

### ISSUE THREAD

**Issue by claytonrothschild** [pchen66/panolens.js#172]
Has anyone been able to correctly read if a device is able to use DeviceOrientationControls?

I would set the default Controls to `DeviceOrientationControls` if the device has sensors, and `OrbitControls` if not.

I've tried a a number of things, mainly:

1.) First check to see if the browser supports window.DeviceOrientationEvent
The idea is that this will != undefined if browser supports. 

Something like:
```
if (window.DeviceOrientationEvent) { 
viewer.setControls(1) }
```
Unfortunately `window.DeviceOrientationEvent ? ` is always != undefined, even if sensors are turned off. This is unexpected based on my reading: https://developers.google.com/web/fundamentals/native-hardware/device-orientation/

2.) Turn on `DeviceOrientationControls` at load, and immediately check ```if (viewer.camera.position.x === 0 && viewer.camera.position.y ===0 && viewer.camera.position.z ===0).```
If its true, the camera is looking straight down (this is common behavior if sensors are off), so revert controls back to `OrbitControls`.

This works like a dream on desktop and Chrome Dev Tools phone emulator. However, on my iPhone the condition is never true, _even though_ the camera itself is indeed looking down. I've seen with other libraries that sometimes devices take time to report initial orientation events, so I tried a 5 second timeout but still no luck.

**TLDR: Has anyone found a reliable way to assess if a device is motion compatible?**

**Comment by kronpano** [comment]
I am using the solution I found here https://www.learnthreejs.com/device-orientation-gyroscope-controls-tutorial/ for my panorama puzzle game (if you want to check if it works https://panopuzzle.created-by.me )  and it works on mobile devices (I know it works on iPhones and most Android phones), tablets and desktop.
In the article is a github link and some explanation.
Have a look - hope it helps

**Comment by claytonrothschild** [comment]
There's really not a lot out there on this. But I found a solution.

Solution 1, above, did not work because indeed every device registers the DeviceOrientationEvent regardless of if controls are available. However, if controls are not available, then after the eventListener is registered it will not actually receive events.

https://stackoverflow.com/a/9856906/7547525

So my final solution was similar to below. All the best.

```
var canHandleOrientation;
if (window.DeviceOrientationEvent) {
    window.addEventListener("deviceorientation", handleOrientation, false);
}

function handleOrientation(event){
  console.log("Orientation:" + event.alpha + ", " + event.beta + ", " + event.gamma);
  canHandleOrientation = event; // will be either null or with event data
}
```
